tests/calculations/test_pw.py

```diff
@@ -250,3 +250,67 @@ def test_pw_parallelization_duplicate_cmdline_flag(fixture_sandbox, generate_cal
     with pytest.raises(InputValidationError) as exc:
         generate_calc_job(fixture_sandbox, entry_point_name, inputs)
     assert 'Conflicting' in str(exc.value)
+
+
+@pytest.mark.parametrize('calculation', ['nscf', 'bands', 'scf', 'relax', 'vc-relax'])
+def test_pw_validate_inputs_restart(
+    fixture_sandbox, generate_calc_job, generate_inputs_pw, fixture_localhost, generate_remote_data, calculation
+):
+    """Test the input validation of restart settings for the ``PwCalculation``."""
+    remote_data = generate_remote_data(computer=fixture_localhost, remote_path='/cat_stevens')
+    entry_point_name = 'quantumespresso.pw'
+
+    inputs = generate_inputs_pw()
+    parameters = inputs['parameters'].get_dict()
+    parameters['CONTROL']['calculation'] = calculation
+
+    if calculation in ('nscf', 'bands'):
```

Comment:
Since you essentially have two types of test, I would just make two separate parametrized tests:
`test_pw_validate_inputs_calculation_base` and `test_pw_validate_inputs_calculation_nscf`